src/Documents/Session/EntityToJson.ts

```diff
@@ -71,19 +71,32 @@ export class EntityToJson {
             return;
         }
 
-        documentInfo.metadata[CONSTANTS.Documents.Metadata.NESTED_OBJECT_TYPES] = 
-            typeInfo.nestedTypes;
-        documentInfo.metadata[CONSTANTS.Documents.Metadata.RAVEN_JS_TYPE] =
-            typeInfo.typeName;
-
         let setMetadata: boolean = false;
-        const metadataNode: object = {};
+        let metadataNode: object = {};
+        
+        documentInfo.metadata[CONSTANTS.Documents.Metadata.NESTED_OBJECT_TYPES] = typeInfo.nestedTypes;
+        documentInfo.metadata[CONSTANTS.Documents.Metadata.RAVEN_JS_TYPE] = typeInfo.typeName;
 
-        if (documentInfo.metadata 
-            && Object.keys(documentInfo.metadata).length > 0) {
+        if (documentInfo.metadataInstance) {

```

Comment:
Can we switch `if` and `else if` blocks so that it's more similar to JVM codebase?

https://github.com/ravendb/ravendb-jvm-client/blob/v4.0/src/main/java/net/ravendb/client/documents/session/EntityToJson.java#L77

First check for `documentInfo.metadata && Object.keys(documentInfo.metadata).length > 0` and then `documentInfo.metadataInstance`.